projects/angular-auth-oidc-client/src/lib/config/config.service.ts

```diff
@@ -112,23 +112,30 @@ export class ConfigurationService {
     const usedConfig = this.prepareConfig(passedConfig);
     this.saveConfig(usedConfig);
 
-    const alreadyExistingAuthWellKnownEndpoints = this.storagePersistenceService.read('authWellKnownEndPoints', usedConfig);
+    const configWithAuthWellKnown = this.addWellKnownEndpoint(usedConfig);
+
+    this.publicEventsService.fireEvent<OpenIdConfiguration>(EventTypes.ConfigLoaded, configWithAuthWellKnown);
+
+    return of(usedConfig);
+  }
+
+  private addWellKnownEndpoint(configuration: OpenIdConfiguration): OpenIdConfiguration {
+    const alreadyExistingAuthWellKnownEndpoints = this.storagePersistenceService.read('authWellKnownEndPoints', configuration);
     if (!!alreadyExistingAuthWellKnownEndpoints) {
-      usedConfig.authWellknownEndpoints = alreadyExistingAuthWellKnownEndpoints;
-      this.publicEventsService.fireEvent<OpenIdConfiguration>(EventTypes.ConfigLoaded, usedConfig);
+      configuration.authWellknownEndpoints = alreadyExistingAuthWellKnownEndpoints;
 
-      return of(usedConfig);
+      return configuration;
```

Comment:
The return type is changed, will this not effect methods using this?

Reply:
The return type did not change, because this is a new private method. The return type of the public method remains the same.